recipes/newrelic/infrastructure/ohi/mysql/debian.yml

```diff
@@ -92,7 +92,13 @@ install:
           EXPECTED_SELECT_REPLICATION=$(echo 'SELECT, REPLICATION')
           EXPECTED_ALL_PRIVILEGES=$(echo 'ALL PRIVILEGES')
           if [ "$CHECK_DB" != "$EXPECTED_SELECT_REPLICATION" ] &&  [ "$CHECK_DB" != "$EXPECTED_ALL_PRIVILEGES" ]; then
-            echo -e "[Error]: Provided user has no access to the desired database.\n - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info."
+            echo "[Error]: Provided user has no access to the desired database. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
+            exit 2
+          fi
+        - |
+          IS_DATABASE_VALID=$(mysql -u{{.NR_CLI_DB_USERNAME}} --port {{.NR_CLI_DB_PORT}} -p{{.NR_CLI_DB_PASSWORD}} -e "use {{.NR_CLI_DATABASE}};SELECT CURDATE()" 2>&1 | grep -v "mysql: [Warning] Using a password on the command line interface can be insecure." | grep 'CURDATE()' | grep -v grep | wc -l)
+          if [ $IS_DATABASE_VALID -eq 0  ]; then
+            echo "[Error]: The provided database name is not accessible with the provided MySQL username and password. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
             exit 2
```

Comment:
Should the exit code be incremented to `3` to be able to distinguish the type of error?

Reply:
I don't think so, it's still in the same step of validating the DB access. We've been pretty consistent on rc 2 for validating the inputs with the specific OHI, rc 1 been no infra-agent.